src/conf_mode/system_option.py

```diff
@@ -138,6 +141,48 @@ def generate(options):
                 f'initcall_blacklist=acpi_cpufreq_init amd_pstate={mode}')
         if 'quiet' in options['kernel']:
             cmdline_options.append('quiet')
+
+    if 'disable_hpet' in kernel_opts:
+        cmdline_options.append('hpet=disable')
+
+    if 'disable_mce' in kernel_opts:
+        cmdline_options.append('mce=off')
+
+    if 'disable_softlockup' in kernel_opts:
+        cmdline_options.append('nosoftlockup')
+
+    # CPU options
+    isol_cpus = k_cpu_opts.get('isolate_cpus')
+    if isol_cpus:
+        cmdline_options.append(f'isolcpus={isol_cpus}')
+
+    nohz_full = k_cpu_opts.get('nohz_full')
+    if nohz_full:
+        cmdline_options.append(f'nohz_full={nohz_full}')
+
+    rcu_nocbs = k_cpu_opts.get('rcu_no_cbs')
+    if rcu_nocbs:
+        cmdline_options.append(f'rcu_nocbs={rcu_nocbs}')
+
+    if 'disable_nmi_watchdog' in k_cpu_opts:
+        cmdline_options.append('nmi_watchdog=0')
+
+    # Memory options
+    if 'disable_numa_balancing' in k_memory_opts:
+        cmdline_options.append('numa_balancing=disable')
+
+    default_hp_size = k_memory_opts.get('default_hugepage_size')
+    if default_hp_size:
+        cmdline_options.append(f'default_hugepagesz={default_hp_size}')
+
+    hp_size = k_memory_opts.get('hugepage_size')
+    if hp_size:
+        cmdline_options.append(f'hugepagesz={hp_size}')
+
+    hp_count = k_memory_opts.get('hugepage_count')
+    if hp_count:
+        cmdline_options.append(f'hugepages={hp_count}')
+
```

Comment:
I may be wrong, but as I understand, the order is important here, and `default_hugepagesz` should be the latest item.

Reply:
According to csit https://github.com/FDio/csit/blob/f6cfb7d3c00c4cc678df3de0b4c12aef97017f93/docs/content/infrastructure/testbed_configuration/sm_icx_hw_bios_cfg.md?plain=1#L1099  they use `default_hugepages` on the first place, should work fine.